controllers/provisioners/eks/update.go

```diff
@@ -116,6 +116,27 @@ func (ctx *EksInstanceGroupContext) UpdateScalingGroup() error {
 		ctx.Log.Info("updated scaling group tags", "instancegroup", instanceGroup.GetName(), "scalinggroup", asgName)
 	}
 
+	isSuspendProcessesUpdateNeeded, processesToSuspend, processesToResume := ctx.SuspendProcessesUpdateNeeded()
```

Comment:
Can this be just a boolean predicate instead?
You can move all the logic of finding which processes to suspend or resume to a method inside `helpers.go` e.g. `UpdateScalingProcesses`.
So that Create/Update calls > UpdateScalingProcesses and it calls > AwsWorker.SetSuspendProcesses or AwsWorker.SetResumeProcesses according to whether there is a diff
Example here:
https://github.com/keikoproj/instance-manager/blob/master/controllers/provisioners/eks/helpers.go#L428